### PR TITLE
retrieve last common build of a module for Docker and Singularity

### PR DESCRIPTION
**Pull request by mirpedrol**
Closes [#1656](https://github.com/nf-core/tools/issues/1656)

When obtaining Docker and Singularity containers from a bioconda package and version, get the latest common build for both containers.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/utils.py [hunk 1]
```diff
@@ -701,16 +701,28 @@ def get_tag_date(tag_date):
...
+                        if match is not None:
+                            all_singularity[match.group(1)] = {"date": get_tag_date(img["updated"]), "image": img}
+                # Obtain common builds from Docker and Singularity images
+                common_keys = list(all_docker.keys() & all_singularity.keys())
+                for k in common_keys:
+                    # Get the most recent common image
+                    if not docker_image or all_docker[k]["date"] > get_tag_date(docker_image["updated"]):
+                        docker_image = all_docker[k]["image"]
+                    if not singularity_image or all_singularity[k]["date"] > get_tag_date(singularity_image["updated"]):
+                        singularity_image = all_singularity[k]["image"]
```
Comment (ErikDanielsson) [inline nf_core/utils.py:725]:
```suggestion
                current_date = datetime.datetime(0, 0, 0, 0, 0)
                for k in common_keys:
                    # Get the most recent common image
                    date = max(get_tag_date(all_docker[k]["date"]]), get_tag_date(all_docker[k]["date"]))
                    if docker_image is None or current_date < date:
                        docker_image = all_docker[k]["image"]
                        singularity_image = all_singularity[k]["image"]
                        current_date = date
```
This should guarantee that we get the same image for both Singularity and Docker.

**Review by ErikDanielsson — COMMENTED** [review]
Looks good, just one minor suggestion

File: nf_core/utils.py (hunk 1 above)
Comment (mirpedrol) [reply nf_core/utils.py]: Thanks for the suggestion!

File: nf_core/utils.py [hunk 2]
```diff
@@ -701,16 +701,30 @@ def get_tag_date(tag_date):
...
+                        # Obtain version and build
+                        match = re.search(r"(?::)+([A-Za-z\d\-_.]+)", img["image_name"])
+                        if match is not None:
+                            all_singularity[match.group(1)] = {"date": get_tag_date(img["updated"]), "image": img}
+                # Obtain common builds from Docker and Singularity images
+                common_keys = list(all_docker.keys() & all_singularity.keys())
+                current_date = None
+                for k in common_keys:
+                    # Get the most recent common image
+                    date = max(get_tag_date(all_docker[k]["date"]), get_tag_date(all_docker[k]["date"]))
```
Comment (ErikDanielsson) [inline nf_core/utils.py:723]:
```suggestion
                    date = max(all_docker[k]["date"], all_docker[k]["date"])
```

**Review by ErikDanielsson — APPROVED** [review]
LGTM! Just add the suggestion (I created a bug with my earlier suggestion ;))

**Pull request merged**